python_ta/transforms/ExprWrapper.py

```diff
@@ -189,6 +192,97 @@ def parse_container_op(
         """Convert an astroid List, Set, Tuple node to a list of z3 expressions."""
         return [self.reduce(element) for element in node.elts]
 
+    def apply_in_op(
+        self, left: z3.ExprRef, right: Union[z3.ExprRef, List[z3.ExprRef], str], negate=False
+    ) -> z3.ExprRef:
+        """
+        Apply `in` or `not in` operator on a list or string and return the
+        resulting z3 expression. Raise Z3ParseException if the operands
+        do not support `in` operator
+        """
+        if isinstance(right, list):  # container type (list/set/tuple)
+            return (
+                z3.And(*[left != element for element in right])
+                if negate
+                else z3.Or(*[left == element for element in right])
+            )
+        elif isinstance(right, (str, z3.SeqRef)):  # string literal or variable
```

Comment:
1. I don't think `right` can actually be a `str`, it should always be a Z3 type. (This is also related to this function's type annotation.)
2. Also check for the type of `left` here

Reply:
<img width="490" alt="image" src="https://github.com/user-attachments/assets/5d7be82b-d231-43e0-a91c-81322472164e">

Hi Professor, based on what I found it seems that `z3.Contains` does support both `str` and `z3.SeqRef`. But of course I will add checking for left-hand-side as well.